test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -213,6 +213,11 @@ object Assertion {
   final def containsString(element: String): Assertion[String] =
     Assertion.assertion("containsString")(param(element))(_.contains(element))
 
+  final def containsTheSameElements[A](other: Iterable[A]): Assertion[Iterable[A]] =
+    Assertion.assertion("containsTheSameElements")(param(other)) { actual =>
+      actual.size == other.size && actual.toSet.diff(other.toSet).isEmpty
```

Comment:
It would be nice to figure out a way to get this assertion to report some more information about which elements are different between the two collections. You could try using `AssertionRec` and having a sub-assertion that the difference is empty. May take a little playing around to be sure we get good error reporting for different combinations of the two collections.

Reply:
Sounds good, let me see how `AssertionRec` that works.